Proxmox VE network configuration: multiple fixes

sbraz opened the pull request:
I split this into multiple commits to make it easier to read and explain the reasoning behind each of them.

Commit:
proxmox-network: remove bond-master line

Proxmox's ifconfig2 does not understand this setting and it is redundant
with bond-slaves.

Commit:
proxmox-network: use spaces for indentation

The doc previously used a mix of tabs and spaces.

Commit:
proxmox-network: use names for bonding options instead of opaque IDs

Commit:
proxmox-network: drop "bond-updelay" option

We do not set it in our other guides. I don't see why we should set it
here.
If we want to add it, it needs to be added everywhere consistently.

Commit:
proxmox-network: stop doing DHCP

Using static IPs is probably better, especially when we're doing
aggregation.

Commit:
proxmox-network: fix typos

Commit:
proxmox-network: do not force a MAC for bonds

ifupdown2 uses the MAC address of the first slave (I tested this).
Therefore, if slaves are declared in the correct order, there is no need
to force a MAC addresss.

Commit:
proxmox-network: use "inet manual" instead of static when no IP is set

I think "inet static" would work but for consistency, let's use manual
everywhere.

Commit:
proxmox-network: use CIDR notation in VM example

It's more modern and easier to read. We already use CIDR everywhere else
in the documentation.

Commit:
proxmox-network: add LACP options to the vRack example

The LACP options were only specified for routed additional IPs, the
vRack example was missing quite a few.

Commit:
proxmox-network: use the same bridge options everywhere

Let's drop the VLAN-related option, it was not used everywhere. Instead,
only keep options that Proxmox sets by default for bridges.

Commit:
proxmox-network: normalize quotes

Commit:
proxmox-network: make both routed and vRack examples more similar

Commit:
proxmox-network: remove declaration of slave interfaces

They are not necessary for the configuration to work.

Commit:
proxmox-network: remove miimon/downdelay bond options

* According to https://docs.kernel.org/networking/bonding.html, 100 is the
  default for miimon, meaning we do not need to force it.
* As for downdelay, it's unlikely that we need to change it from its
  default of 0. The value of this option was very likely copied from
  existing config samples found on the Internet.

Commit:
proxmox-network: drop duplicate "auto vmbr0" line

Commit:
proxmox-network: move the IP address from bond0 to vmbr0 in routed mode

Otherwise, the configuration doesn't work:
```
utils.py:344:is_addr_ip_allowed_on(): info: bond0: ignoring ip address. Assigning an IP address is not allowed on enslaved interfaces. bond0 is enslaved to vmbr0
```
Also move the definition of vmbr0 above that of private interfaces.

Commit:
proxmox-network: make ADV configuration closer to SCALE/HGR

The previous configuration didn't have the ethernet interface in the
bridge, which was weird.
Let's also configure a secondary private bridge to match the SCALE/HGR
config sample.

Commit:
proxmox-network: fix wording

Commit:
proxmox-network: move HGR/SCALE vRack to a tab

Commit:
proxmox-network: document additional IPs via vRack for ADV - SK-1686

Commit:
proxmox-network: rework sections and descriptions

This commit doesn't change the configuration examples, I just rephrased
and reorganised the text while fixing typos.

Commit:
proxmox-network: add VM example configuration for Netplan via vRack

sbraz commented:
I'm done refactoring the doc. I have tested the configuration on an ADV and a SCALE server. @gary-ovh, can you please test the configuration yourself too? I can grant you access to my servers if that helps, they are both attached to the same vRack and each of them has an additional IP routed to them.
There is a rather large commit ("rework sections and descriptions") because I had many sentences to change and I didn't know how to split it properly.

Commit:
proxmox-network: make Debian VM examples more similar

remicmacs commented on pages/bare_metal_cloud/dedicated_servers/proxmox-network-HG-Scale/guide.en-gb.md:
````diff
@@ -370,19 +370,35 @@ systemctl restart networking.service
+> Debian (ifupdown)
+>> Contents of the `/etc/network/interfaces` file:
+>>
+>> ```bash
````
```text

remicmacs commented on pages/bare_metal_cloud/dedicated_servers/proxmox-network-HG-Scale/guide.en-gb.md:
````diff
@@ -324,75 +294,129 @@ To configure the first usable IP address, you must edit the network configuratio
+> Debian (ifupdown)
+>> Contents of the `/etc/network/interfaces` file:
+>>
+>> ```bash
````
```suggestion
>> ```text
```
According to the previous modifications

remicmacs reviewed (approved):
LGTM but as previously said on this topic, such changes can only be really validated via a lab, I think

sbraz commented:
> LGTM but as previously said on this topic, such changes can only be really validated via a lab, I think

I did test the configuration on an ADV Gen 3 server and a HGR/SCALE one. Also @gary-ovh told me he would test the configuration as well.

Commit:
proxmox-network: mention that the guide doesn't apply to ADV-Gen2 and older

Commit:
proxmox-network: use title case for range names

The website doesn't use SCALE or ADVANCE in uppercase.

gary-ovh commented:
Thanks @sbraz! I tested all configs and everything is working perfectly!

@Y0Coss / @remicmacs this is ready for your review/translations

Y0Coss committed:
EN minor adjustments + EN>FR translation

sbraz commented on pages/bare_metal_cloud/dedicated_servers/proxmox-network-HG-Scale/guide.fr-fr.md:
```diff
@@ -16,7 +18,7 @@ updated: 2024-10-18
 ## Prérequis
 
 - Disposer d'un [serveur dédié OVHcloud](/links/bare-metal/bare-metal)
```
J'enlèverais "disposer" du premier élément de la liste, ou alors on le remet sur le second

sbraz commented on pages/bare_metal_cloud/dedicated_servers/proxmox-network-HG-Scale/guide.fr-fr.md:
```diff
@@ -33,273 +35,243 @@ Avec cette configuration, les adresses Additional IP doivent être attachées à
-- ajouter les routes.
+- créer une interface d'agrégation (*bond*) (uniquement pour les gammes High Grade & Scale) ;
+- créer une interface bridge connectée au *bond* ;
+- autoriser le transfert de paquets entre les interfaces ;
```
- autoriser le routage de paquets entre les interfaces ;

sbraz commented on pages/bare_metal_cloud/dedicated_servers/proxmox-network-HG-Scale/guide.fr-fr.md:
```diff
@@ -33,273 +35,243 @@ Avec cette configuration, les adresses Additional IP doivent être attachées à
+>>         # Ajoutez un bloc IP
+>>         up ip route add ADDITIONAL_IP_BLOCK/28 dev $IFACE
+>>
 >> # Private
```
Est-ce qu'on traduit Private ?

Y0Coss replied:
Traduit et adapté aussi en EN

Commit:
proxmox-network: small changes to the en/fr guides

I've tried to use "block of Additional IPs" or the French equivalent in
most sentences to really highlight the name of the product.

Y0Coss committed:
Last adjustments

Y0Coss committed:
Missing translation

Y0Coss committed:
Duplication

sbraz commented on pages/bare_metal_cloud/dedicated_servers/proxmox-network-HG-Scale/guide.en-gb.md:
```diff
@@ -350,13 +350,13 @@ ssh PUB_IP_DEDICATED_SERVER
 >> iface lo inet loopback
 >>
->> # Public
+>> # Public interfaces
```
Attention dans ADV il n'y a qu'une interface publique et une privée donc pas de pluriel

Y0Coss committed:
Missing translation + Advance wording adjustments

Y0Coss commented:
Thank you for your contribution @sbraz
